pkg/cpu/cpu_darwin.go

```diff
@@ -0,0 +1,110 @@
+package cpu
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+var sysctlOutput = make(map[string]string)
+
+func (i *Info) load() error {
+	err := populateSysctlOutput()
+	if err != nil {
+		return errors.Wrap(err, "unable to populate sysctl map")
+	}
+
+	i.TotalCores = getTotalCores()
+	i.TotalThreads = getTotalThreads()
+	i.Processors = getProcessors()
+
+	return nil
+}
+
+// getProcessors some more info https://developer.apple.com/documentation/kernel/1387446-sysctlbyname/determining_system_capabilities
+func getProcessors() []*Processor {
+	p := make([]*Processor, getProcTopCount())
+	for i, _ := range p {
+		p[i] = new(Processor)
+		p[i].Vendor = sysctlOutput[fmt.Sprintf("hw.perflevel%s.name", strconv.Itoa(i))]
+		p[i].Model = getVendor()
+		p[i].NumCores = getNumberCoresFromPerfLevel(i)
+		p[i].Capabilities = getCapabilities()
```

Comment:
Would it be possible to construct the `ghw.Processor` struct with a c[ollection of `ghw.ProcessorCore` structs](https://github.com/jaypipes/ghw/blob/835f6a233b6db0d384a7e5a254eef4f05cdcae5e/pkg/cpu/cpu.go#L63)?

You can look into this in a future PR if you'd like, but to equate MacOS support with Linux support, I'd like to do this.

Reply:
Hey @jaypipes I was thinking about using the `ProcessorCore` but wasn't too sure how to integrate it!

Just to recap...in ARM we have the p cores and e cores which I basically split into two `Processor` entries. The content looks like this (notice p[i].Vendor)
<img width="668" alt="Screenshot 2023-12-31 at 12 25 18 PM" src="https://github.com/jaypipes/ghw/assets/19938874/a40920d1-ce55-46bf-bd8a-8eef0beb3a16">


And what this is what is looks like I run the README code 
```
cpu (2 physical packages, 10 cores, 10 hardware threads)
 physical package #0 (8 cores, 0 hardware threads)
  capabilities: [FEAT_FlagM FEAT_SHA3 FEAT_SB armv8_crc32 FEAT_LSE2 FEAT_JSCVT
                 FEAT_DotProd FEAT_SHA512 FEAT_SHA256 FEAT_PAuth FEAT_CSV2 FEAT_RDM
                 FEAT_SSBS FEAT_CSV3 FEAT_DIT FEAT_LRCPC FEAT_LSE FEAT_FP16
                 FEAT_SHA1 FEAT_LRCPC2 AdvSIMD_HPFPCvt FEAT_FRINTTS FEAT_FlagM2 FEAT_FCMA
                 FEAT_FHM AdvSIMD floatingpoint]
 physical package #0 (2 cores, 0 hardware threads)
  capabilities: [FEAT_AES FEAT_PMULL FEAT_DPB2 FEAT_FlagM FEAT_SHA3 FEAT_SB
                 armv8_crc32 FEAT_LSE2 FEAT_JSCVT FEAT_DotProd FEAT_SHA512 FEAT_SHA256
                 FEAT_PAuth FEAT_CSV2 FEAT_RDM FEAT_SSBS FEAT_CSV3 FEAT_DIT
                 FEAT_LRCPC FEAT_LSE FEAT_FP16 FEAT_SHA1 FEAT_LRCPC2 AdvSIMD_HPFPCvt
                 FEAT_FRINTTS AdvSIMD floatingpoint]
```

`ProcessorCore` got all the `int` attributes like IDs and other fun stuff which can read from `/proc` `/sys` but seems like they are not really applicable for macOS (atleast in ARM will need to check intel)?


To start with...for some parity should I just create some PC entries based on the total number of cores? e.g
```
p[i].Cores = make([]*ProcessorCore, getTotalCores())
```

Reply:
@pixelsoccupied I'm very sorry for not responding on this pull request. For some reason, this just slipped from my radar.

Yes, absolutely, just create a slice of empty ProcessorCore structs for now, that's totally fine.

Reply:
Added!